runtime/src/iree/hal/drivers/cuda/cuda_allocator.c

```diff
@@ -24,6 +24,7 @@ typedef struct iree_hal_cuda_allocator_t {
   iree_hal_cuda_context_wrapper_t* context;
   CUdevice device;
   CUstream stream;
+  iree_hal_cuda_memory_pools_t* pools;
```

Comment:
I'd assume eventually we'll allocate from this `pools` for this "unpooled" allocator too?

Reply:
That's my thought! we can make the pools arbitrarily complex (add as many as we want, segregate resources by lifetime, based on usage requirements as export across process requires a different type of pool, etc). I've got a few fixes incoming that switch nearly all our allocations besides variables/constants and whatever the user decides to do to queue-ordered so hopefully our usage of the base allocator will be for weird stuff (import/export, etc)!